cou/steps/plan.py

```diff
@@ -349,15 +349,61 @@ def generate_common_plan(
                 raise_on_blocked=True,
             ),
         )
-    )
+    ]
     if args.backup:
-        plan.add_step(
+        steps.append(
             PreUpgradeStep(
                 description="Backup mysql databases",
                 coro=backup(analysis_result.model),
             )
         )
-    return plan
+
+    return steps
+
+
+def _get_post_upgrade_steps(analysis_result: Analysis, args: CLIargs) -> list[PostUpgradeStep]:
+    """Get the post-upgrade steps.
+
+    :param analysis_result: Analysis result
+    :type analysis_result: Analysis
+    :param args: CLI arguments
+    :type args: CLIargs
+    :return: List of post-upgrade steps.
+    :rtype: list[PreUpgradeStep]
+    """
+    steps = []
+    if args.upgrade_group in {DATA_PLANE, None}:
+        steps.append(_get_ceph_mon_post_upgrade_step(analysis_result))
+
+    return steps
+
+
+def _get_ceph_mon_post_upgrade_step(analysis_result: Analysis) -> PostUpgradeStep:
```

Comment:
nit: if we were using `analysis_result` to access more things I think it would make sense to pass as parameter. In this case I think it can be control_plane_apps directly